src/rust/node-identifier/src/node_identifier.rs

```diff
@@ -0,0 +1,234 @@
+use std::collections::HashMap;
+
+use failure::Error;
+use grapl_utils::rusoto_ext::dynamodb::GraplDynamoDbClientExt;
+use rust_proto_new::graplinc::grapl::api::graph::v1beta1::{
+    GraphDescription,
+    IdentifiedGraph,
+    IdentifiedNode,
+    NodeDescription,
+};
+use tap::tap::TapOptional;
+
+use crate::{
+    dynamic_sessiondb::NodeDescriptionIdentifier,
+    error::NodeIdentifierError,
+};
+
+/**
+    The `NodeIdentifier` takes in graphs of previously unidentified nodes and
+    performs identification based on the configured strategies for that node
+    type.
+
+    The strategies come in two variants:
+
+    * [Session](`graplinc::grapl::api::graph::v1beta1::Session`) - strategy used
+      for nodes with lifetimes (e.g. process, network connection)
+
+    * [Static](`graplinc::grapl::api::graph::v1beta1::Static`) - strategy used
+      for nodes with canonical and unique identifiers (e.g. aws events)
+*/
+#[derive(Clone)]
+pub(crate) struct NodeIdentifier<D>
+where
+    D: GraplDynamoDbClientExt,
+{
+    dynamic_identifier: NodeDescriptionIdentifier<D>,
+}
+
+impl<D> NodeIdentifier<D>
+where
+    D: GraplDynamoDbClientExt,
+{
+    pub(crate) fn new(dynamic_identifier: NodeDescriptionIdentifier<D>) -> Self {
+        Self { dynamic_identifier }
+    }
+
+    // todo: We should be yielding IdentifiedNode's here
+    #[tracing::instrument(fields(node_key=?node.node_key), skip(self, node))]
+    async fn attribute_node_key(&self, node: &NodeDescription) -> Result<IdentifiedNode, Error> {
+        let new_node = self.dynamic_identifier.attribute_dynamic_node(node).await?;
+        Ok(new_node.into())
+    }
```

Comment:
feel like this todo is very important and easily missed; perhaps a ticket or something (or you saying 'yep im doing that ASAP next thing')

Reply:
This was a consequence of porting existing code. I'm not sure what this comment actually means, unfortunately.